tests/testsuite/features2.rs

```diff
@@ -1114,29 +1294,38 @@ fn proc_macro_ws() {
         .build();
 
     p.cargo("check -p pm -v")
-        .with_stderr_contains("[RUNNING] `rustc --crate-name foo [..]--cfg[..]feat1[..]")
+        .with_stderr_data(str![[r#"
+[LOCKING] 2 packages to latest compatible versions
+[CHECKING] foo v0.1.0 ([ROOT]/foo/foo)
+[RUNNING] `rustc --crate-name foo [..]--cfg[..]feat1[..]`
+[CHECKING] pm v0.1.0 ([ROOT]/foo/pm)
+[RUNNING] `rustc --crate-name pm [..]--crate-type proc-macro[..]`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
         .run();
     // This may be surprising that `foo` doesn't get built separately. It is
     // because pm might have other units (binaries, tests, etc.), and so the
     // feature resolver must assume that normal deps get unified with it. This
     // is related to the bigger issue where the features selected in a
     // workspace depend on which packages are selected.
     p.cargo("check --workspace -v")
-        .with_stderr(
-            "\
-[FRESH] foo v0.1.0 [..]
-[FRESH] pm v0.1.0 [..]
-[FINISHED] `dev` profile [..]
-",
-        )
+        .with_stderr_data(str![[r#"
+[FRESH] foo v0.1.0 ([ROOT]/foo/foo)
+[FRESH] pm v0.1.0 ([ROOT]/foo/pm)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
         .run();
     // Selecting just foo will build without unification.
     p.cargo("check -p foo -v")
         // Make sure `foo` is built without feat1
-        .with_stderr_line_without(
-            &["[RUNNING] `rustc --crate-name foo --edition=2015"],
-            &["--cfg[..]feat1"],
-        )
+        .with_stderr_data(str![[r#"
+[CHECKING] foo v0.1.0 ([ROOT]/foo/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 foo/src/lib.rs --error-format=json --json=[..] --crate-type lib --emit=[..] -C embed-bitcode=no -C debuginfo=2 -C split-debuginfo=unpacked --check-cfg[..] --check-cfg[..] -C metadata=[..] -C extra-filename=[..] --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
```

Comment:
The `line_without` should be analogous to `does_not_contain`, but the tracking issue doesn't mention this function.